rye/src/cli/init.rs

```diff
@@ -291,3 +393,244 @@ pub fn execute(cmd: Args) -> Result<(), Error> {
+    if let Some(section) = config.get("metadata") {
+        if metadata.name.is_none() {
+            if let Some(Some(name)) = section.get("name") {
+                metadata.name = Some(name.to_string());
+            }
+        }
+        if metadata.version.is_none() {
+            if let Some(Some(version)) = section.get("version") {
+                metadata.version = Some(version.to_string());
+            }
+        }
+        if metadata.description.is_none() {
+            if let Some(Some(description)) = section.get("description") {
+                metadata.description = Some(description.to_string());
+            }
+        }
+        if metadata.author.is_none() {
+            if let Some(Some(author)) = section.get("author") {
+                let email = match section.get("author_email") {
+                    Some(Some(it)) => it,
+                    _ => "",
+                };
+                metadata.author = Some((author.to_string(), email.to_string()));
+            }
+        }
+        if metadata.license.is_none() {
+            if let Some(Some(license)) = section.get("license") {
+                metadata.license = Some(license.to_string());
+            }
+        }
+    }
+    if let Some(section) = config.get("options") {
+        if metadata.requires_python.is_none() {
+            if let Some(Some(python_requires)) = section.get("python_requires") {
+                metadata.requires_python = Some(python_requires.to_string());
+            }
+        }
+        if let Some(Some(reqs)) = section.get("install_requires") {
+            reqs.lines()
+                .filter_map(|x| Requirement::from_str(x).ok())
+                .for_each(|x| {
+                    requirements.insert(x.name.to_string(), format_requirement(&x).to_string());
+                });
+        }
+    }
+    Ok(())
+}
+
+fn get_setup_py_json<T: AsRef<Path>>(path: T, python: T) -> Result<Value, Error> {
+    let python = python.as_ref();
+    let setup_py = path.as_ref();
+    let temp_dir = tempdir()?;
+    let dir = setup_py
+        .parent()
+        .context("could not establish setup.py parent dir")?;
+
+    let options = CopyDirOptions {
+        exclude: vec![dir.join(".git"), dir.join(".tox")],
+    };
+    copy_dir(dir, temp_dir.path(), &options)?;
+
+    let setuptools_proxy = temp_dir.path().join("setuptools.py");
+    fs::write(setuptools_proxy, SETUP_PY_PROXY_SCRIPT)?;
+
+    let cmd = Command::new(python)
+        .arg(setup_py)
+        .env("PYTHONPATH", temp_dir.path())
+        .stderr(Stdio::piped())
+        .output()?;
+    if cmd.status.success() {
+        Ok(serde_json::from_slice(&cmd.stderr)?)
+    } else {
+        let log = String::from_utf8_lossy(&cmd.stderr);
+        bail!("failed to proxy setup.py\n{}", log);
+    }
+}
+
+/// See https://github.com/konstin/poc-monotrail/blob/7487250e5ace3447f25a5573b7a9953ccd9537e/src/requirements_txt.rs#L12-L16
```

Comment:
It looks like the link here has a 404 status.

Reply:
Nice catch